akka-projection-dynamodb-integration/src/test/scala/akka/projection/dynamodb/CatchupSpec.scala

```diff
@@ -0,0 +1,185 @@
+
+  val config: Config = ConfigFactory
+    .parseString(s"""
+    akka.projection.dynamodb.replay-on-rejected-sequence-numbers=off
+    akka.projection {
+      restart-backoff {
+        min-backoff = 20 ms
+        max-backoff = 2 s
+      }
+    }
+    """)
+    .withFallback(TestConfig.config)
+
+  final case class Processed(projectionId: ProjectionId, envelope: EventEnvelope[String])
+
+  class FailingTestHandler(
+      projectionId: ProjectionId,
+      probe: ActorRef[Processed],
+      processedEvents: ConcurrentHashMap[String, java.lang.Boolean],
+      failEvents: ConcurrentHashMap[String, Int])
+      extends Handler[EventEnvelope[String]] {
+    private val log = LoggerFactory.getLogger(getClass)
+
+    override def process(envelope: EventEnvelope[String]): Future[Done] = {
+      val failCount = failEvents.getOrDefault(envelope.event, 0)
+      if (failCount > 0) {
+        failEvents.put(envelope.event, failCount - 1)
+        log.debug(
+          "{} Fail event [{}], pid [{}], seqNr [{}]",
+          projectionId.key,
+          envelope.event,
+          envelope.persistenceId,
+          envelope.sequenceNr)
+        throw TestException(s"Fail event [${envelope.event}]")
+      } else {
+        log.debug(
+          "{} Processed {} [{}], pid [{}], seqNr [{}]",
+          projectionId.key,
+          if (processedEvents.containsKey(envelope.event)) "duplicate event" else "event",
+          envelope.event,
+          envelope.persistenceId,
+          envelope.sequenceNr)
+        val wasAbsent = processedEvents.putIfAbsent(envelope.event, true) == null
+        // duplicates are ok but not reported to probe
+        if (wasAbsent)
+          probe ! Processed(projectionId, envelope)
+        Future.successful(Done)
+      }
+    }
+  }
+
+}
+
+class CatchupSpec
+    extends ScalaTestWithActorTestKit(CatchupSpec.config)
+    with AnyWordSpecLike
+    with TestDbLifecycle
+    with TestData
+    with BeforeAndAfterAll
+    with LogCapturing {
+  import CatchupSpec._
+
+  override def typedSystem: ActorSystem[_] = system
+
+  private val log = LoggerFactory.getLogger(getClass)
+
+  private val entityType = nextEntityType()
+  private val sliceRange = 0 to 1023
+  private val projectionId = genRandomProjectionId()
+
+  private def slice(pid: PersistenceId): Int = Persistence(system).sliceForPersistenceId(pid.id)
+
+  def spawnAtLeastOnceProjection(handler: Handler[EventEnvelope[String]]): ActorRef[ProjectionBehavior.Command] = {
+    val sourceProvider =
+      EventSourcedProvider
+        .eventsBySlices[String](system, DynamoDBReadJournal.Identifier, entityType, sliceRange.min, sliceRange.max)
+
+    spawn(
+      ProjectionBehavior(
+        DynamoDBProjection
+          .atLeastOnce(projectionId, settings = None, sourceProvider = sourceProvider, handler = () => handler)))
+  }
+
+  "A Projection" must {
+    "catchup old events without rejections and replays" in {
+      pending // FIXME there are rejections
```

Comment:
Porting it to dynamodb, and it actually fails here with rejections. I'll have to dig into this.